website/client/js/services/pusherService.js

```diff
@@ -59,29 +59,49 @@ angular.module('habitrpg')
       });
 
       // When the user correctly enters the party channel
-      partyChannel.bind('pusher:subscription_succeeded', function(members) {
-        // TODO members = [{id, info}]
-
-        // If we just reconnected after some inactivity, sync the party
-        if (reconnecting === true) {
-          Groups.party(true).then(function (syncedParty) {
-            // Assign and not replace so that all the references get the modifications
-            if ($rootScope.party) {
-              _.assign($rootScope.party, syncedParty);
-              $rootScope.loadingParty = false; // make sure the party is set as loaded
+      partyChannel.bind('pusher:subscription_succeeded', function(pusherMembers) {
+        // Wait for the party to be loaded
+        Groups.party(reconnecting ? true : false).then(function (party) {
```

Comment:
Though, to be honest, I don't think `Groups.party` cares that it is a `Boolean` just whether or not it's truthy. So `Groups.party(reconnecting)` should work.